### PR TITLE
shadowsocks-rust: update to 1.10.1

### PR DESCRIPTION
**Pull request by eatradish**
Topic Description
-----------------

shadowsocks-rust: update to 1.10.1

- Enable local-http-native-tls in loongson3 and ppc64el to fix build
- Use rust nightly to build on loongson3 and ppc64el to fix build

Package(s) Affected
-------------------

shadowsocks-rust: 1.10.1

Security Update?
----------------

No

Architectural Progress
----------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

File: extra-network/shadowsocks-rust/autobuild/loongson3/prepare [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export SODIUM_USE_PKG_CONFIG=1
+
+abinfo "Setting HOME as /root to fix rustup ..."
+export HOME="/root"
+
+abinfo "Setting rust version as nightly to fix build ..."
+curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain nightly -y
+export PATH=$HOME/.cargo/bin:$PATH
```
Comment (MingcongBai) [inline extra-network/shadowsocks-rust/autobuild/loongson3/prepare]: Quote your variable.

File: extra-network/shadowsocks-rust/autobuild/loongson3/prepare [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+source $HOME/.cargo/env
```
Comment (MingcongBai) [inline extra-network/shadowsocks-rust/autobuild/loongson3/prepare]: Ditto.

File: extra-network/shadowsocks-rust/autobuild/defines [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
 
-FAIL_ARCH="!(amd64|arm64)"
 USECLANG=1
 ABTYPE=rust
 ABSPLITDBG=0
+
+CARGO_AFTER__PPC64EL="--features local-http-native-tls"
+CARGO_AFTER__LOONGSON3="--features local-http-native-tls"
+
+NOLTO__PPC64EL=1
```
Comment (MingcongBai) [inline extra-network/shadowsocks-rust/autobuild/defines]: Why?

File: extra-network/shadowsocks-rust/autobuild/ppc64el/prepare [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export SODIUM_USE_PKG_CONFIG=1
+
+abinfo "Setting HOME as /root to fix rustup ..."
+export HOME="/root"
+
+abinfo "Setting rust version as nightly to fix build ..."
+curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain nightly -y
+export PATH=$HOME/.cargo/bin:$PATH
```
Comment (MingcongBai) [inline extra-network/shadowsocks-rust/autobuild/ppc64el/prepare]: Okay. You could have not used architectural sub-directories here (especially not by making a copy for each architecture - come on...!), I'd suggest that you use a big `if` branch instead.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-network/shadowsocks-rust/autobuild/defines [hunk 5]
```diff
@@ -1,10 +1,15 @@
...
+BUILDDEP="asciidoc pkg-config llvm"
+
 PKGDES="A Rust port of shadowsocks, lightweight yet secured SOCKS5 proxy"
 
-FAIL_ARCH="!(amd64|arm64)"
 USECLANG=1
 ABTYPE=rust
 ABSPLITDBG=0
+
+# Since ring does not support ppc64el and loongson3 architecture, use local-http-native-tls to avoid this dependency
```
Comment (MingcongBai) [inline extra-network/shadowsocks-rust/autobuild/defines]: Architectures *

File: extra-network/shadowsocks-rust/autobuild/prepare [hunk 6]
```diff
@@ -1 +1,8 @@
 export SODIUM_USE_PKG_CONFIG=1
+	
+abinfo "Setting HOME as /root to fix rustup ..."
+export HOME="/root"
+
+abinfo "Setting rust version as nightly to fix build ..."
+curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain nightly -y
+source $HOME/.cargo/env
```
Comment (MingcongBai) [inline extra-network/shadowsocks-rust/autobuild/prepare]: Quote your variable.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Comment by eatradish** [comment]
All LGTM

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**